miniz_oxide/src/inflate/core.rs

```diff
@@ -672,6 +672,138 @@ fn start_static_table(r: &mut DecompressorOxide) {
+    146800640, 2294284288, 1220542464, 3368026112, 683671552, 2831155200, 1757413376, 3904897024,
+    415236096, 2562719744, 1488977920, 3636461568, 952107008, 3099590656, 2025848832, 4173332480,
+    79691776, 2227175424, 1153433600, 3300917248, 616562688, 2764046336, 1690304512, 3837788160,
+    348127232, 2495610880, 1421869056, 3569352704, 884998144, 3032481792, 1958739968, 4106223616,
+    213909504, 2361393152, 1287651328, 3435134976, 750780416, 2898264064, 1824522240, 3972005888,
+    482344960, 2629828608, 1556086784, 3703570432, 1019215872, 3166699520, 2092957696, 4240441344,
+    46137344, 2193620992, 1119879168, 3267362816, 583008256, 2730491904, 1656750080, 3804233728,
+    314572800, 2462056448, 1388314624, 3535798272, 851443712, 2998927360, 1925185536, 4072669184,
+    180355072, 2327838720, 1254096896, 3401580544, 717225984, 2864709632, 1790967808, 3938451456,
+    448790528, 2596274176, 1522532352, 3670016000, 985661440, 3133145088, 2059403264, 4206886912,
+    113246208, 2260729856, 1186988032, 3334471680, 650117120, 2797600768, 1723858944, 3871342592,
+    381681664, 2529165312, 1455423488, 3602907136, 918552576, 3066036224, 1992294400, 4139778048,
+    247463936, 2394947584, 1321205760, 3468689408, 784334848, 2931818496, 1858076672, 4005560320,
+    515899392, 2663383040, 1589641216, 3737124864, 1052770304, 3200253952, 2126512128, 4273995776,
+    29360128, 2176843776, 1103101952, 3250585600, 566231040, 2713714688, 1639972864, 3787456512,
+    297795584, 2445279232, 1371537408, 3519021056, 834666496, 2982150144, 1908408320, 4055891968,
+    163577856, 2311061504, 1237319680, 3384803328, 700448768, 2847932416, 1774190592, 3921674240,
+    432013312, 2579496960, 1505755136, 3653238784, 968884224, 3116367872, 2042626048, 4190109696,
+    96468992, 2243952640, 1170210816, 3317694464, 633339904, 2780823552, 1707081728, 3854565376,
+    364904448, 2512388096, 1438646272, 3586129920, 901775360, 3049259008, 1975517184, 4123000832,
+    230686720, 2378170368, 1304428544, 3451912192, 767557632, 2915041280, 1841299456, 3988783104,
+    499122176, 2646605824, 1572864000, 3720347648, 1035993088, 3183476736, 2109734912, 4257218560,
+    62914560, 2210398208, 1136656384, 3284140032, 599785472, 2747269120, 1673527296, 3821010944,
+    331350016, 2478833664, 1405091840, 3552575488, 868220928, 3015704576, 1941962752, 4089446400,
+    197132288, 2344615936, 1270874112, 3418357760, 734003200, 2881486848, 1807745024, 3955228672,
+    465567744, 2613051392, 1539309568, 3686793216, 1002438656, 3149922304, 2076180480, 4223664128,
+    130023424, 2277507072, 1203765248, 3351248896, 666894336, 2814377984, 1740636160, 3888119808,
+    398458880, 2545942528, 1472200704, 3619684352, 935329792, 3082813440, 2009071616, 4156555264,
+    264241152, 2411724800, 1337982976, 3485466624, 801112064, 2948595712, 1874853888, 4022337536,
+    532676608, 2680160256, 1606418432, 3753902080, 1069547520, 3217031168, 2143289344, 4290772992,
+];
```

Comment:
Would it be possible to generate this table at compile time using `const` functions?

Reply:
Unfortunately [not](https://github.com/rust-lang/rust/issues/93481). I did try to construct this using `const fn`, but even for a reduced version it runs into the linked issue.

Reply:
Looking into this a bit more, it _is_ possible:

```rs
static REVERSED_BITS_LOOKUP: [u32; 1024] = {
    let mut table = [0; 1024];

    let mut i = 0;
    while i < 1024 {
        table[i] = (i as u32).reverse_bits();
        i += 1;
    }

    table
};
```

Presumably my previous attempts had an issue of some sort. I will have to see if this crate's MSRV supports loops in const fn.

Reply:
Support was added in [1.46.0](https://blog.rust-lang.org/2020/08/27/Rust-1.46.0.html#const-fn-improvements).